File: src/AlgorithmOfEuclid.java
public class AlgorithmOfEuclid {
    public static void main(String[] args) {
        int result = gcd(10, 15);
        System.out.println(result);
    }

    public static int gcd(int p, int q) {
        if (q == 0) return p;
        int r = p % q;
        return gcd(q, r);
    }
}

File: src/algorithm_of_euclid/AlgorithmOfEuclid.kt
package algorithm_of_euclid

fun main() {
    val result = gcd(10, 15)
    println(result)
}

private fun gcd(p: Int, q: Int): Int {
    if (q == 0) return p
    val r = p % q
    return gcd(q, r)
}